### PR TITLE
Fix #2283: Added New Haptic Feedback Behavior

### LINKED ISSUE #2283: Add New Haptic Feedback Behavior
Add haptic feedback:

- [x] Selecting sponsored image logo
- [x] Selecting image credit button on background images.
- [x] Switching tabs

Internal Chats: 
[one](https://bravesoftware.slack.com/archives/C06UXF3KJ/p1580333633059600) [two](https://bravesoftware.slack.com/archives/C06UXF3KJ/p1580333661060200)

### PR DESCRIPTION
**Pull request by jhreis**
## Summary of Changes

*Note to Merger*:
Please add merge description:
> New feedback for:
> - Switching tabs (including swiping bottom toolbar)
> - Sponsored image logo selection
> - Selecting image credit for normal backgrounds

This pull request fixes issue #2283 

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Test Plan:

- Sponsored Image:

1. Load a new tab that is a sponsored image.
1. Click on the sponsor's logo
1. Should feel haptic feedback / vibration

- Background Image Credit:

1. Load a normal background tab.
1. Click on the photographer's name.
1. Should feel haptic feedback / vibration (and see menu come up)

- Tab Switching:

> Any time a new tab is focused, you should feel the haptic feedback, some cases to test:

- [ ] Creating a new tab (this is focusing a new tab)
- [ ] Deleting a tab (this is focusing a new tab, since current one is gone)
- [ ] Deleting the _last_ tab (have one tab open, delete it via the long-press menu on tab tray)
- [ ] Swiping the bottom toolbar to navigate between tabs
- [ ] Opening the tab tray and creating a new tab
- [ ] Opening the tab tray and selecting a different tab

## Reviewer Checklist:

- [x] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `release-notes/(include|exclude)`
  - `bug` / `enhancement`
- [x] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [x] Adequate unit test coverage exists to prevent regressions.
- [x] Adequate test plan exists for QA to validate (if applicable).
- [x] Issue is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: Shared/Extensions/ImpactFeedbackGeneratorExtensions.swift [hunk 1]
```diff
@@ -0,0 +1,15 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import UIKit
+
+extension UIImpactFeedbackGenerator {
+    @discardableResult
+    public func bzzt() -> UIImpactFeedbackGenerator {
```
Comment (kylehickinson) [inline Shared/Extensions/ImpactFeedbackGeneratorExtensions.swift]: I believe you can just use `Self` here but it doesn't really matter :)

**Review by kylehickinson — APPROVED** [review]
### BZZT

**Comment by jhreis** [comment]
Maybe I should name the function 📳

**Comment by iccub** [comment]
Rebase please

**Comment by srirambv** [comment]
Even with vibrate set on both ring and silent, I don't see any haptic feedback on either SI or regular image. Nothing on tab switch either.

**Comment by jhreis** [comment]
@srirambv still not triggering for you?

**Comment by srirambv** [comment]
@jhreis all good. I had to change the settings which I had forgotten. Issue has been verified and updated with notes :+1:

**Pull request merged**